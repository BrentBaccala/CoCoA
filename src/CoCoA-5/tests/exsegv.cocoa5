Use R;

----------------------------------------------------------------------
PrintLn "ex segm fault for error with multiline list input 10-05-2011";
Try
  gcd([2,
      "b"]);
UponError e Do  PrintLn GetErrMesg(e);
EndTry
  
----------------------------------------------------------------------
PrintLn "ex segm fault for error on args in NewMat 30-01-2012";
Try  ZeroMat(R,2,"a"); UponError e Do PrintLn GetErrMesg(e); EndTry
Try  ZeroMat(2,3,2); UponError e Do PrintLn GetErrMesg(e); EndTry
Try  ZeroMat(R,"a",2); UponError e Do PrintLn GetErrMesg(e); EndTry

----------------------------------------------------------------------
PrintLn "ex segm fault on first and last on short lists 04-07-2012";
Try  first([]); UponError e Do PrintLn GetErrMesg(e); EndTry
Try  last([]);  UponError e Do PrintLn GetErrMesg(e); EndTry
Try  first([],2);  UponError e Do PrintLn GetErrMesg(e); EndTry
Try  last([],2);   UponError e Do PrintLn GetErrMesg(e); EndTry

----------------------------------------------------------------------
PrintLn "\nStill alive!";
----------------------------------------------------------------------

-- Try // changed interface: args are indets instead of indices
--   PrintLn "ex segm fault indices out of range in ContentWRT/CoefficientsWRT";
--   ***ContentWRT(xyz-xz^2, [-2])***;
-- UponError e Do
--   PrintLn GetErrMesg(e); PrintLn;
-- EndTry

