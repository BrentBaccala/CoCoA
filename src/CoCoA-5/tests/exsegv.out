
# 
# 
# 
# ex segm fault for error with multiline list input 10-05-2011

# 
Try # 
Try # 
Try # 
UponError # A list containing RINGELEM, INT or RAT values is required

# 
# 
# ex segm fault for error on args in NewMat 30-01-2012

# Expecting type INT, but found type STRING

# Expecting type RING, but found type INT

# Expecting type INT, but found type STRING

# 
# 
# ex segm fault on first and last on short lists 04-07-2012

# list is empty

# list is empty

# value greater than list length

# value greater than list length

# 
# 
# 
Still alive!

# 
# 
# 
# 
# 
# 
# 
# 
# 
# 